src/raster/rasterband.rs

```diff
@@ -600,3 +609,143 @@ impl ColorInterpretation {
         _string(rv)
     }
 }
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum PaletteInterpretation {
+    Gray,
+    Rgba,
+    Cmyk,
+    Hls,
+}
+
+impl PaletteInterpretation {
+    fn from_c_int(palette_interpretation: GDALPaletteInterp::Type) -> Self {
+        match palette_interpretation {
+            GDALPaletteInterp::GPI_Gray => Self::Gray,
+            GDALPaletteInterp::GPI_RGB => Self::Rgba,
+            GDALPaletteInterp::GPI_CMYK => Self::Cmyk,
+            GDALPaletteInterp::GPI_HLS => Self::Hls,
+            _ => unreachable!("GDAL has implemented a new type of `GDALPaletteInterp`"),
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct GrayEntry {
+    pub g: i16,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct RgbaEntry {
+    pub r: i16,
+    pub g: i16,
+    pub b: i16,
+    pub a: i16,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct CmykEntry {
+    pub c: i16,
+    pub m: i16,
+    pub y: i16,
+    pub k: i16,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct HlsEntry {
+    pub h: i16,
+    pub l: i16,
+    pub s: i16,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum ColorEntry {
+    Gray(GrayEntry),
+    Rgba(RgbaEntry),
+    Cmyk(CmykEntry),
+    Hls(HlsEntry),
+}
+
+/// Color table for raster bands that use the PaletteIndex color interpretation.
+///
+/// This object carries the lifetime of the raster band that
+/// contains it. This is necessary to prevent the raster band
+/// from being dropped before the color table.
+pub struct ColorTable<'a> {
+    palette_interpretation: PaletteInterpretation,
+    c_color_table: GDALColorTableH,
+    _raster_band: &'a RasterBand<'a>,
+}
+
+impl<'a> ColorTable<'a> {
+    fn from_c_color_table(c_color_table: GDALColorTableH, raster_band: &'a RasterBand<'a>) -> Self {
+        let interp_index = unsafe { gdal_sys::GDALGetPaletteInterpretation(c_color_table) };
+        ColorTable {
+            palette_interpretation: PaletteInterpretation::from_c_int(interp_index),
+            c_color_table,
+            _raster_band: raster_band,
+        }
+    }
+
+    /// How the values of this color table are interpreted.
+    pub fn palette_interpretation(&self) -> PaletteInterpretation {
+        self.palette_interpretation
+    }
+
+    /// Get the number of color entries in this color table.
+    pub fn entry_count(&self) -> i32 {
```

Comment:
@ChristianBeilschmidt 
Do you think this should return a `usize`? Same question for the `index` parameter of `entry` and `entry_as_rgb`.

Reply:
Very good point.
For some reason, `layer_count` and `raster_count` on `Dataset` both return 'isize`.
Does maybe @jdroenner know why?

Does GDAL at some point output a negative number? We can't put the whole size range into a function that requires int on the GDAL side, but actually an unsigned type makes more sense.

Reply:
`GDALGetColorEntryCount` returns an `int` but I don't see how it could be negative. The documentation simply says "Get number of color entries in table". Maybe the safest thing is to leave it as `i32`?

Reply:
For me, it should be an unsigned int in a Rust API. One could argue that if we have `i32` with only positive numbers, we should do something like `u16`. But the _Rusty_ way is `usize`.

But maybe there is some convention for this API that it uses ints for this. Maybe someone else has an opinion for this.

Reply:
I think that I'll go ahead and make it `usize`. I can change it back it it's decided otherwise.